### PR TITLE
[Latest posts] Fixes the categories selector crash when category does not exist

### LINKED ISSUE #20915: [ Latest posts ] Category filter crashes block
**Describe the bug**
When a category does not exist and you try to filter by it the block crashes.

**To reproduce**
1. Make sure you have posts and/or pages assigned to categories
2. Add a new page
3. Add a `LatestPosts` block
4. Try to write in the `Categories` filter random text and press ENTER
5. Block crashes

**Expected behavior**
Nothing should happen if the category I typed does not exist or a message should be displayed.

**Screenshots**
category-crash

cc: @Ringish

### PR DESCRIPTION
**Pull request by draganescu**
## Description
Closes  #20915

## How has this been tested?
Tested locally by:

- make sure you have posts in categories
- add a page or post
- insert a LatestPosts block
- try to filter by a category that does not exist
- nothing should happen

## Types of changes
Non breaking changes to the LatestPosts edit component.

cc @Ringish

### CODE REVIEW

**Comment by aduth** [comment]
I notice #20961 also aims to solve this. @draganescu @Ringish Preference on which to keep?

**Comment by draganescu** [comment]
@aduth thanks! :) we know, in a weird sync we both published the PRs at the same time. Well decide which is better.

File: packages/block-library/src/latest-posts/edit.js [hunk 1]
```diff
@@ -121,9 +121,18 @@ class LatestPostsEdit extends Component {
...
-			);
+			const allCategories = tokens.map( ( token ) => {
+				const newToken =
+					typeof suggestions[ token ] !== 'undefined'
+						? suggestions[ token ]
+						: null;
+				return typeof token === 'string' ? newToken : token;
+			} );
+			// we do nothing if the category is not selected
+			// from suggestions
```
Comment (talldan) [inline packages/block-library/src/latest-posts/edit.js]:
```suggestion
			// We do nothing if the category is not selected
			// from suggestions.
```

**Review by talldan — APPROVED** [review]
Worked for me, though I did find the code a little hard to follow. I wondered if an early return like this might be more preferable at the start of the selectCategories function. Then the mapping function wouldn't have to worry about returning `null`:
```js
			const hasNoSuggestion = tokens.some(
				( token ) => typeof token === 'string' && ! suggestions[ token ]
			);

			if ( hasNoSuggestion ) {
				return;
			}
```
